src/Processors/Transforms/DistinctTransform.cpp

```diff
@@ -76,6 +182,18 @@ void DistinctTransform::transform(Chunk & chunk)
     for (auto pos : key_columns_pos)
         column_ptrs.emplace_back(columns[pos].get());
 
+    bool has_lc_mask{false};
+    IColumn::Filter lc_mask;
```

Comment:
Better to make it a pointer, something like `std::unique_ptr<IColumn::filter>` and remove `has_lc_mask`.